src/normalizer.ts

```diff
@@ -313,42 +273,38 @@ class V1SchemaNormalizer {
       exports: this.exports,
       imports: this.imports,
       schemas: this.schemas,
+      errors: this.errors,
+      warnings: this.warnings,
     }
   }
 
-  querySchemaRef(ref: string, path: string[]): Schema | null {
+  // NOTE: this $ref should already be validated as pointing to a schema in the parser
+  querySchemaRef(ref: string): Schema {
     const parts = ref.split('/')
-    if (parts[0] !== '#' || parts[1] !== 'components' || parts[2] !== 'schemas') {
-      this.recordError("Not a valid ref " + ref, path);
-      return null;
-    }
-
-    const name = parts[3];
-    const s = this.schemas[name]
-    if (!s) {
-      const availableSchemas = Object.keys(this.schemas).join(', ')
-      this.recordError(`Invalid reference ${ref}. Cannot find schema ${name}. Options are: [${availableSchemas}]`, path);
-      return null;
-    }
-
-    return s
+    const name = parts[3]
+    return this.schemas[name]!
   }
 
-  annotateType(s: any, path: string[]): XtpNormalizedType | undefined {
+  annotateType(s: any): XtpNormalizedType | undefined {
     if (!s || typeof s !== 'object' || Array.isArray(s)) return undefined
     if (s.xtpType) return s.xtpType
 
-    if ((s.type && s.type === 'object') ||
-      (s.properties && s.properties.length > 0)) {
+    if (s.type && s.type === 'object' && s.additionalProperties) {
```

Comment:
we validated the shape of this in the parser so now we're just looking for the pattern